src/dbos.ts

```diff
@@ -1279,10 +1279,10 @@ export class DBOS {
   static startWorkflow<T extends object>(targetClass: T, params?: StartWorkflowParams): InvokeFunctionsAsync<T>;
   static startWorkflow(
     target: ((...args: unknown[]) => Promise<unknown>) | ConfiguredInstance | object,
-    params?: StartWorkflowParams,
+    params?: StartWorkflowFunctionParams<unknown>,
   ): unknown {
-    const instance = typeof target === 'function' ? null : (target as ConfiguredInstance);
-    if (instance && !(instance instanceof ConfiguredInstance)) {
+    const instance = typeof target === 'function' ? params?.instance : (target as ConfiguredInstance);
```

Comment:
I actually meant to remove `StartWorkflowFunctionParams`. I don't think we should support the function overload of `startWorkflow` with instance methods. It's so easy to screw up using `this` unbound. And even if you get it right, it's a poor developer experience compared to the `ConfiguredInstance` variant of `startWorkflow`. 

I have an idea for marking instance methods on decoration/registration so that we can throw in startWorkflow if you try and use an unbound instance method.